pandas/_libs/lib.pyx

```diff
@@ -902,7 +902,8 @@ def indices_fast(ndarray index, const int64_t[:] labels, list keys,
                     val = keys[j][sorted_labels[j][i - 1]]
                     PyTuple_SET_ITEM(tup, j, val)
                     Py_INCREF(val)
-
+                if len(tup) == 1:
+                    tup = tup[0]
```

Comment:
We could yes. We would have to put lines 900-904 into an if else block, same for lines 911-915. I found this a bit better to read, but could change it, if this would be preffered?

Reply:
i think i like the if/else block, be sure to add a comment on why is going on

Reply:
Thanks for the quick answer, is if else now.